engine/class_modules/sc_mage.cpp

```diff
@@ -2275,9 +2276,15 @@ struct mage_spell_t : public spell_t
       trigger_frostfire_mastery();
 
     // Needs to be triggered with a delay so that ABar doesn't eat its own proc
-    if ( !background && harmful && rng().roll( p()->talents.intuition->effectN( 1 ).percent() ) )
-      make_event( *sim, [ this ] { p()->buffs.intuition->trigger(); } );
-  }
+    if ( !background && harmful ) {
+      if ( p()->buffs.intuition_tracker->at_max_stacks() ||  rng().roll( p()->talents.intuition->effectN( 1 ).percent() )  ) {
```

Comment:
Bunch of code style violations. `{` goes on a new line, there should be just a single space around operators (you have two after `||`), etc.